src/components/UsersPage/UsersPage.tsx

```diff
@@ -1,24 +1,42 @@
+import { useState } from 'react';
+
 import { usePreviewContext } from '../../contexts/previewContext';
 import { pages } from '../../hooks/useRouter';
 import { trpc } from '../../trpc/trpcClient';
 import { Link } from '../Link';
-import { LayoutMain } from '../LayoutMain';
+import { LayoutMain, PageContent } from '../LayoutMain';
+import { UsersPageFiltersPanel } from '../UsersPageFilterPanel/UsersPageFilterPanel';
+import { CommonHeader } from '../CommonHeader';
+import { UserFilterQuery } from '../../modules/userTypes';
 
 import { tr } from './UsersPage.i18n';
 
 export const UsersPage = () => {
-    const usersQuery = trpc.user.getList.useQuery({});
+    const [searchQuery, setSearchQuery] = useState('');
+    const [filtersQuery, setFiltersQuery] = useState<UserFilterQuery>({});
+    const usersQuery = trpc.user.getList.useQuery({ search: searchQuery, ...filtersQuery });
+
     const { showUserPreview } = usePreviewContext();
 
     return (
         <LayoutMain pageTitle={tr('Users')}>
-            {usersQuery.data?.map((user) => (
-                <div key={user.id}>
-                    <Link onClick={() => showUserPreview(user.id)} href={pages.user(user.id)}>
-                        {user.name}
-                    </Link>
-                </div>
-            ))}
+            <CommonHeader title={tr('Users')} />
+            <UsersPageFiltersPanel
+                filterState={filtersQuery}
+                total={usersQuery.data?.total || 0}
+                counter={usersQuery.data?.counter || 0}
+                setSearchQuery={setSearchQuery}
+                onFilterApply={setFiltersQuery}
+            />
+            <PageContent>
+                {usersQuery.data?.users.map((user) => (
+                    <div key={user.id}>
+                        <Link onClick={() => showUserPreview(user.id)} href={pages.user(user.id)}>
```

Comment:
Lets use UserListItem here

Reply:
I don't think UserListItem suits here, I think it's better to make a new component in next commit

Reply:
ok